### DISCUSSION TITLE
Handling events in testing web driver io  and node js

### DISCUSSION THREAD

**Discussion by vinodkrp** [webdriverio/webdriverio#6727] [Q&A]
Hi All,

I need help on preparing end to end testing of the events which are added using the event listeners using web driver io.

Can someone please help me on this issue?

**Comment by christian-bromann** [comment]
@vinodkrp not sure which events are you interested in but you can use the [`getPuppeteer`](https://webdriver.io/docs/api/browser/getPuppeteer/) command to use the DevTools API for listening to page events.
